File: Linked_List/basic_ll.java
package Linked_List;
import java.util.*;
class Node{
    int data;
    Node next;
    Node(int data){
        this.data=data;
        this.next=null;
    }
}
public class basic_ll {
    public static Node insert(Node head,int data){
        Node newNode=new Node(data);
        if(head==null){
            return newNode;
        }
        Node temp=head;
        while(temp.next!=null){
            temp=temp.next;
        }
        temp.next=newNode;
        return head;
    }
    public static void printList(Node head){
        Node temp=head;
        while(temp!=null){
            System.out.print(temp.data + "->");
            temp=temp.next;
        }
    }
    public static void main(String[] args) {
        Node head=null;
        head=insert(head,10);
        head=insert(head,20);
        head=insert(head,30);

        printList(head);
    }
}

File: Linked_List/basic_LL.cpp
#include <iostream>
using namespace std;

class Node {
public:
    int data;
    Node* next;

    Node(int data) {
        this->data = data;
        next = NULL;
    }
};

// Insert at end
Node* insert(Node* head, int data) {

    Node* newNode = new Node(data);

    // If list is empty
    if (head == NULL) {
        return newNode;
    }

    Node* temp = head;

    // Move to last node
    while (temp->next != NULL) {
        temp = temp->next;
    }

    // Connect last node to new node
    temp->next = newNode;

    return head;
}

// Print linked list
void printList(Node* head) {

    Node* temp = head;

    while (temp != NULL) {
        cout << temp->data << "->";
        temp = temp->next;
    }
}

int main() {

    Node* head = NULL;

    // Insert elements
    head = insert(head, 10);
    head = insert(head, 20);
    head = insert(head, 30);

    // Print list
    printList(head);

    return 0;
}